@potiuk any suggestions that I can fix this https://github.com/apache/airflow/runs/5648830794?check_suite_focus=true#step:11:1039

since there is not `DEFAULT_FORMAT_PREFIX` in 2.1.1 https://github.com/apache/airflow/blob/2.1.1/airflow/utils/operator_helpers.py#L20

```
There were errors when importing Providers!
This test is run to check Provider's compatibility with Airflow 2.1.0
So this error might mean that your providers are not Airflow 2.1.0 compatible

Detailed errors:

----------------------------------------
Exception when importing: airflow.providers.google.cloud.example_dags.example_presto_to_gcs


Traceback (most recent call last):
  File "/opt/airflow/dev/import_all_classes.py", line 90, in import_all_classes
    _module = importlib.import_module(modinfo.name)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name, package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/google/cloud/example_dags/example_presto_to_gcs.py", line 32, in <module>
    from airflow.providers.google.cloud.transfers.presto_to_gcs import PrestoToGCSOperator
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/google/cloud/transfers/presto_to_gcs.py", line 24, in <module>
    from airflow.providers.presto.hooks.presto import PrestoHook
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/presto/hooks/presto.py", line 31, in <module>
    from airflow.utils.operator_helpers import AIRFLOW_VAR_NAME_FORMAT_MAPPING, DEFAULT_FORMAT_PREFIX
ImportError: cannot import name 'DEFAULT_FORMAT_PREFIX' from 'airflow.utils.operator_helpers' (/usr/local/lib/python3.7/site-packages/airflow/utils/operator_helpers.py)
```

--- a/airflow/providers/presto/hooks/presto.py
+++ b/airflow/providers/presto/hooks/presto.py
@@ -15,6 +15,7 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+import json
 import os
 import warnings
 from typing import Any, Callable, Iterable, Optional, overload
@@ -27,6 +28,34 @@ from airflow import AirflowException
 from airflow.configuration import conf
 from airflow.hooks.dbapi import DbApiHook
 from airflow.models import Connection
+from airflow.utils.operator_helpers import AIRFLOW_VAR_NAME_FORMAT_MAPPING
+
+try:
+    from airflow.utils.operator_helpers import DEFAULT_FORMAT_PREFIX
+except ImportError:
+    # This is from airflow.utils.operator_helpers,
+    # For the sake of provider backward compatibility, this is hardcoded if import fails
+    # https://github.com/apache/airflow/pull/22416#issuecomment-1075531290
+    DEFAULT_FORMAT_PREFIX = 'airflow.ctx.'
+
+
+def generate_presto_client_info() -> str:
+    """Return json string with dag_id, task_id, execution_date and try_number"""
+    context_var = {
+        format_map['default'].replace(DEFAULT_FORMAT_PREFIX, ''): os.environ.get(
+            format_map['env_var_format'], ''
+        )
+        for format_map in AIRFLOW_VAR_NAME_FORMAT_MAPPING.values()
+    }
+    task_info = {
+        'dag_id': context_var['dag_id'],
+        'task_id': context_var['task_id'],
+        'execution_date': context_var['execution_date'],
+        'try_number': context_var['try_number'],
+        'dag_run_id': context_var['dag_run_id'],
+        'dag_owner': context_var['dag_owner'],
+    }
+    return json.dumps(task_info, sort_keys=True)
 
 
 class PrestoException(Exception):
@@ -83,11 +112,13 @@ class PrestoHook(DbApiHook):
                 ca_bundle=extra.get('kerberos__ca_bundle'),
             )
 
+        http_headers = {"X-Presto-Client-Info": generate_presto_client_info()}
         presto_conn = prestodb.dbapi.connect(
             host=db.host,
             port=db.port,
             user=db.login,
             source=db.extra_dejson.get('source', 'airflow'),
+            http_headers=http_headers,
             http_scheme=db.extra_dejson.get('protocol', 'http'),
             catalog=db.extra_dejson.get('catalog', 'hive'),
             schema=db.schema,

--- a/tests/providers/presto/hooks/test_presto.py
+++ b/tests/providers/presto/hooks/test_presto.py
@@ -28,7 +28,31 @@ from prestodb.transaction import IsolationLevel
 
 from airflow import AirflowException
 from airflow.models import Connection
-from airflow.providers.presto.hooks.presto import PrestoHook
+from airflow.providers.presto.hooks.presto import PrestoHook, generate_presto_client_info
+
+
+def test_generate_airflow_presto_client_info_header():
+    env_vars = {
+        'AIRFLOW_CTX_DAG_ID': 'dag_id',
+        'AIRFLOW_CTX_EXECUTION_DATE': '2022-01-01T00:00:00',
+        'AIRFLOW_CTX_TASK_ID': 'task_id',
+        'AIRFLOW_CTX_TRY_NUMBER': '1',
+        'AIRFLOW_CTX_DAG_RUN_ID': 'dag_run_id',
+        'AIRFLOW_CTX_DAG_OWNER': 'dag_owner',
+    }
+    expected = json.dumps(
+        {
+            "dag_id": "dag_id",
+            "execution_date": "2022-01-01T00:00:00",
+            "task_id": "task_id",
+            "try_number": "1",
+            "dag_run_id": "dag_run_id",
+            "dag_owner": "dag_owner",
+        },
+        sort_keys=True,
+    )
+    with patch.dict('os.environ', env_vars):
+        assert generate_presto_client_info() == expected
 
 
 class TestPrestoHookConn(unittest.TestCase):
@@ -45,6 +69,7 @@ class TestPrestoHookConn(unittest.TestCase):
             catalog='hive',
             host='host',
             port=None,
+            http_headers=mock.ANY,
             http_scheme='http',
             schema='hive',
             source='airflow',
@@ -98,6 +123,7 @@ class TestPrestoHookConn(unittest.TestCase):
             catalog='hive',
             host='host',
             port=None,
+            http_headers=mock.ANY,
             http_scheme='http',
             schema='hive',
             source='airflow',
@@ -116,6 +142,51 @@ class TestPrestoHookConn(unittest.TestCase):
             sanitize_mutual_error_response=True,
             service_name='TEST_SERVICE_NAME',
         )
+        assert mock_connect.return_value == conn
+
+    @patch('airflow.providers.presto.hooks.presto.generate_presto_client_info')
+    @patch('airflow.providers.presto.hooks.presto.prestodb.auth.BasicAuthentication')
+    @patch('airflow.providers.presto.hooks.presto.prestodb.dbapi.connect')
+    @patch('airflow.providers.presto.hooks.presto.PrestoHook.get_connection')
+    def test_http_headers(
+        self,
+        mock_get_connection,
+        mock_connect,
+        mock_basic_auth,
+        mocked_generate_airflow_presto_client_info_header,
+    ):
+        mock_get_connection.return_value = Connection(
+            login='login', password='password', host='host', schema='hive'
+        )
+        client = json.dumps(
+            {
+                "dag_id": "dag-id",
+                "execution_date": "2022-01-01T00:00:00",
+                "task_id": "task-id",
+                "try_number": "1",
+                "dag_run_id": "dag-run-id",
+                "dag_owner": "dag-owner",
+            },
+            sort_keys=True,
+        )
+        http_headers = {'X-Presto-Client-Info': client}
+
+        mocked_generate_airflow_presto_client_info_header.return_value = http_headers['X-Presto-Client-Info']
+
+        conn = PrestoHook().get_conn()
+        mock_connect.assert_called_once_with(
+            catalog='hive',
+            host='host',
+            port=None,
+            http_headers=http_headers,
+            http_scheme='http',
+            schema='hive',
+            source='airflow',
+            user='login',
+            isolation_level=0,
+            auth=mock_basic_auth.return_value,
+        )
+        mock_basic_auth.assert_called_once_with('login', 'password')
         assert mock_connect.return_value == conn
 
     @parameterized.expand(

